src/net_processing.cpp

```diff
@@ -2760,17 +2760,15 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // fProvidesHeaderAndIDs is used to "lock in" version of compact blocks we send (fWantsCmpctWitness)
         if (!State(pfrom.GetId())->fProvidesHeaderAndIDs) {
             State(pfrom.GetId())->fProvidesHeaderAndIDs = true;
-            State(pfrom.GetId())->fWantsCmpctWitness = nCMPCTBLOCKVersion == 2;
+            State(pfrom.GetId())->fWantsCmpctWitness = true;
         }
-        if (State(pfrom.GetId())->fWantsCmpctWitness == (nCMPCTBLOCKVersion == 2)) { // ignore later version announces
+        if (State(pfrom.GetId())->fWantsCmpctWitness) {
```

Comment:
We could've previously fallen into this if `fWantsCmpctWitness == false && nCMPCTBLOCKVersion != 2`, but we won't now. I'm not sure what the consequence of that is.

Reply:
This if block is handling subsequent `sendcmpct` messages from the peer. The idea is that we'll ignore any subsequent `sendcmpct` message where the version doesn't match the version in the first received `sendcmpct` message, i.e. if we receive:

```
sendcmpct false 2
sendcmpct true 2
```

Then the first `sendcmpct` announces support for compact blocks and the second `sendcmpct` enables high-bandwidth mode. However, if we receive:

```
sendcmpct false 2
sendcmpct true 1
```

the second `sendcmpct` is ignored and we don't enable high-bandwidth mode.

This PR is removing all support for `sendcmcpt <true|false> 1`, so this specialized logic is no longer required.

Reply:
Yes, I see now. Thanks.